api_test_utils/oauth_helper.py

```diff
@@ -72,20 +77,47 @@ async def get_authenticated_with_simulated_auth(self):
             self.base_uri, self.client_id, self.redirect_uri
         )
         return await authenticator.authenticate()
-
-    async def get_authenticated_with_mock_auth(
-        self, user: str = "9999999999", webdriver_session: WebDriver = None
+    
+    def get_authenticated_with_mock_auth(
+        self, user: str = "9999999999"
     ) -> str:
-        """Get the code parameter value required to post to the oauth /token endpoint"""
-        if not webdriver_session:
-            raise Exception(
-                'Cannot run simulated auth without a webdriver session, please import the "webdriver_session" fixture.'
-            )
+        session = requests.Session()
+        resp = session.get(
+            f"{self.base_uri}/authorize",
+            params={
+                "client_id": self.client_id,
+                "redirect_uri": self.redirect_uri,
+                "response_type": "code",
+                "state": "1234567890",
+            },
+        )
 
-        authenticator = _RealAuthFlow(
-            self.base_uri, self.client_id, self.client_secret, self.redirect_uri
+        if resp.status_code != 200:
+            print(json.dumps(resp.json(), indent=2))
+            sys.exit(1)
```

Comment:
Remove the sys.exit(1) and replace with an exception. It's fine in a script, but we don't want pytest to terminate without doing teardown.

Reply:
Replaced with RuntimeError